src/datasources/exchange-api/exchange-api.module.ts

```diff
@@ -4,10 +4,29 @@ import { ExchangeApi } from './exchange-api.service';
 import { IExchangeApi } from '../../domain/interfaces/exchange-api.interface';
 import { ValidationErrorFactory } from '../errors/validation-error-factory';
 import { JsonSchemaService } from '../../common/schemas/json-schema.service';
+import { IConfigurationService } from '../../common/config/configuration.service.interface';
+
+const baseUrlFactory = {
+  provide: 'EXCHANGE_SERVICE_BASE_URL',
```

Comment:
Unfortunately it seems that Nest does not support the `provide` to be a `Symbol` at this level (so I had to fallback to `string` as the identifier)